Makefile

```diff
@@ -44,10 +43,15 @@ default:
 	@echo
 	@exit 1
 
-.state/env/pyvenv.cfg: requirements/dev.txt requirements/docs.txt requirements/lint.txt requirements/ipython.txt
-	# Create our Python 3.8 virtual environment
+.state/01bootstrap: requirements/bootstrap.txt
```

Comment:
How does this not make a circular dep:
`.state/01bootstrap: requirements/bootstrap.txt`
and below:
`requirements/%.txt: .state/01bootstrap requirements/%.in`

?

Reply:
I guess this is not a dependency if there is no corresponding `bootstrap.in` file, but I am not `make` expert. I haven't found the syntax how to exclude `bootstrap.txt` from `%.txt` matching.

Reply:
I've tested that circular dependency doesn't trigger when `boot.in` doesn't exist.

`Makefile`.
```make
default: *.txt

bootstrap: boot.txt

%.txt: bootstrap %.in
        echo $^ $@
```

The test.
```
$ make        
echo bootstrap xxx.in xxx.txt
bootstrap xxx.in xxx.txt
$ touch boot.in
$ make
make: Circular bootstrap <- boot.txt dependency dropped.
echo bootstrap boot.in boot.txt
bootstrap boot.in boot.txt
echo bootstrap xxx.in xxx.txt
bootstrap xxx.in xxx.txt
```

But as you may see, there is a problem with specifying `bootstrap` (or .state/env/pyvenv.cfg`) dependency as it gets passed to the underlying command,

Reply:
Found out how to fix the problem.
```diff
-%.txt: bootstrap %.in
-        echo $^ $@
+%.txt: %.in bootstrap
+        echo $< $@
```
`$^` substitutes all dependencies, `$<` selects only the first dependency ([ref](https://stackoverflow.com/questions/41330138/makefile-special-characters-meaning-and-use/41330163#41330163)).

Now I am the `make` expert. :D

Reply:
Fixed the bug in the latest commit. Thanks for review. Now I see that my requirements files are erased by `pip-compile` when regenerated, but I had no idea what is going on.